src/main/scala/com/fulcrumgenomics/util/ParIterator.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2022 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.util
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.commons.async.AsyncIterator
+
+import java.util.concurrent.ForkJoinPool
+import scala.collection.AbstractIterator
+import scala.collection.parallel.{ForkJoinTaskSupport, TaskSupport}
+
+/**
+  * Companion object to help with manufacturing ParIterators.
+  */
+object ParIterator {
+  /**
+    * Constructs a ParIterator from the underlying iterator, that will perform operations in parallel on chunks
+    * of `chunkSize` using `threads` threads.
+    */
+  def apply[A](iter: Iterator[A], chunkSize: Int = 1024, threads: Int): ParIterator[A] = {
+    val pool = new ForkJoinPool(threads, ForkJoinPool.defaultForkJoinWorkerThreadFactory, null, true)
+    val support = new ForkJoinTaskSupport(pool)
+    apply(iter, chunkSize, support)
+  }
+
+  /**
+    * Constructs a ParIterator from the underlying iterator, that will perform operations in parallel on chunks
+    * of `chunkSize` using the provided `TaskSupport`.
+    */
+  def apply[A](iter: Iterator[A], chunkSize: Int, support: TaskSupport): ParIterator[A] = {
+    val grouped = iter.grouped(chunkSize)
+    val async = AsyncIterator(grouped, Some(2))
```

Comment:
It's possible that this should be optional, but it makes a huge different to have the next chunk ready to go when the parallel operation on the prior chunk wraps up.  Cut runtime by ~1/3 and finally got utilization up to what you'd hope with the number of threads.